src/main/java/reactor/netty/transport/ProxyProvider.java

```diff
@@ -324,6 +340,65 @@ public ProxyProvider build() {
 		}
 	}
 
+	static final class RegexShouldProxyPredicate implements Predicate<SocketAddress> {
+
+		private final String regex;
+		private Pattern pattern;
+
+		/**
+		 * Create a {@link RegexShouldProxyPredicate} based off the provided regular expression.
+		 *
+		 * @param regex The string regular expression
+		 * @return a predicate whether we should direct to proxy
+		 */
+		public static RegexShouldProxyPredicate fromRegularExpression(String regex) {
+			return new RegexShouldProxyPredicate(regex);
+		}
+
+		/**
+		 * Creates a {@link RegexShouldProxyPredicate} based off the provided pattern with possible wildcards as
+		 * described in https://docs.oracle.com/javase/7/docs/api/java/net/doc-files/net-properties.html
+		 *
+		 * @param pattern The string wildcarded expression
+		 * @return a predicate whether we should direct to proxy
+		 */
+		public static RegexShouldProxyPredicate fromWildcardedPattern(String pattern) {
+			return new RegexShouldProxyPredicate(pattern.replaceAll("\\.", "\\\\.").replaceAll("\\*", "\\.*"));
+		}
+
+		private RegexShouldProxyPredicate(String pattern) {
+			this.regex = pattern;
+		}
+
+		private Pattern getPattern() {
+			if (pattern == null) {
+				pattern = Pattern.compile(regex, Pattern.CASE_INSENSITIVE);
+			}
```

Comment:
Let's compile the pattern when the object is created. When there is a regex this mean that we will always check it so no need of lazy compilation of the regex.